Exiled.Events/Patches/Generic/StaminaUsage.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="StaminaUsage.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Generic
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features;
+    using HarmonyLib;
+    using NorthwoodLib.Pools;
+    using PlayerStatsSystem;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="StaminaStat.CurValue"/>.
+    /// </summary>
+    [HarmonyPatch(typeof(StaminaStat), nameof(StaminaStat.CurValue), MethodType.Setter)]
```

Comment:
Also, i dont think that it is a good idea to prevent all sets.
Like, by default float value == 0f, and stamina always be empty

Reply:
Also we can't patch `StaminaStat.CurValue` since `CurValue` doesn't get overridden by `StaminaStat`.